pointcloud/basic_octree.go

```diff
@@ -1,92 +1,94 @@
-package octree
+package pointcloud
 
 import (
-	"context"
-
-	"github.com/edaniels/golog"
 	"github.com/golang/geo/r3"
 	"github.com/pkg/errors"
-
-	pc "go.viam.com/rdk/pointcloud"
 )
 
 const (
+	internalNode = NodeType(iota)
+	leafNodeEmpty
+	leafNodeFilled
+	octreeVersion = 1.0
 	// This value allows for high level of granularity in the octree while still allowing for fast access times
 	// even on a pi.
 	maxRecursionDepth = 1000
 )
 
-// basicOctree is a data structure that represents a basic octree structure with information regarding center
-// point, side length and node data.
-type basicOctree struct {
-	logger     golog.Logger
+// NodeType represents the possible types of nodes in an octree.
+type NodeType uint8
+
+// BasicOctree is a data structure that represents a basic octree structure with information regarding center
+// point, side length and node data. An octree is a data structure that recursively partitions 3D space into
+// octants to represent occupancy. It is a storage format for a pointcloud that allows for better searchability
+// and serialization.
+type BasicOctree struct {
 	node       basicOctreeNode
 	center     r3.Vector
 	sideLength float64
 	size       int
-	meta       pc.MetaData
+	meta       MetaData
 }
 
 // basicOctreeNode is a struct comprised of the type of node, children nodes (should they exist) and the pointcloud's
 // PointAndData datatype representing a point in space.
 type basicOctreeNode struct {
 	nodeType NodeType
-	children []*basicOctree
-	point    pc.PointAndData
+	children []*BasicOctree
```

Comment:
Is your vote for a dictionary/map? Since users dont have access to the children except through the defined functions we should have to worry about corruption in the number but could go with 
`children map[int]*BasicOctree{}` if needed

Reply:
I was thinking more of a fixed size array: `[8]*BasicOctree`. I imagine this doesn't really matter in practice, but I was just curious if there was a reason why `children` needs to be a dynamically sized data structure, given my understanding of an octree is that each node always has 8 children.

Not requesting a change at all, just asking for my own edification.